### DISCUSSION TITLE
Functionality to use continuous outcome with AttentionMIL?

### DISCUSSION THREAD

**Discussion by bplee** [jamesdolezal/slideflow#339] [Q&A] [answered]
Just as the title states, I'm looking to see if there is an easy way to predict a continuous outcome with AttentionMIL using slideflows MIL training functionality. It would seem as if the loss functions as they are implemented in mil_config from slideflow.mil are fixed properties and can't be readily changed to something like MSE.

**Comment by jamesdolezal** [comment, marked as answer]
The short answer is not yet, but it's coming in our next feature release (~3-4 weeks).

The longer answer is that MIL training is built around the FastAI framework, which provides a flexible interface for customizing model training. The function [`sf.mil.build_fastai_learner`](https://slideflow.dev/mil_module/#slideflow.mil.build_fastai_learner) returns a FastAI [Learner](https://docs.fast.ai/learner.html#learner) object, which can be used for customized training. The process of customizing the FastAI learner has not yet been included in our documentation, but might look something like:

```python
# Build a Learner
learner = sf.mil.build_fastai_learner(...)

# Change the loss function
learner.loss_func = ...

# Fit the model
learner.fit(n_epoch=10, lr=1e-4, wd=1e-5)
```

We do plan to add a documentation section dedicated to use and customization of the underlying FastAI Learner, which will provide a great deal of flexibility to researchers wanting to customize aspects of training. This expanded documentation will also be included in the next feature release.

So stay tuned for further developments - apologies that it's not available out of the box right now.

**Reply by bplee** [reply]
Thanks James for such a quick response! I appreciate you pointing me in the right direction, this should be more than enough for me to chew on to try to get something going, and I'll keep my ears peeled for the next release :) Thank you for all the work that you've done in developing and updating this awesome library
